platform/macos/app/MapDocument.m

```diff
@@ -177,6 +177,66 @@ - (NSURL *)shareURL {
 
 #pragma mark File methods
 
+- (IBAction)import:(id)sender {
+    NSOpenPanel *panel = [NSOpenPanel openPanel];
+    panel.allowedFileTypes = @[@"public.json", @"json", @"geojson"];
+    panel.allowsMultipleSelection = YES;
+    
+    __weak __typeof__(self) weakSelf = self;
+    [panel beginSheetModalForWindow:self.window completionHandler:^(NSModalResponse result) {
+        if (result != NSFileHandlingPanelOKButton) {
+            return;
+        }
+        
+        for (NSURL *url in panel.URLs) {
+            [weakSelf importFromURL:url];
+        }
+    }];
+}
+
+/**
+ Adds the contents of the GeoJSON file at the given URL to the map.
+ 
+ GeoJSON features are styled according to
+ [simplestyle-spec](https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0/).
+ */
+- (void)importFromURL:(NSURL *)url {
+    MGLStyle *style = self.mapView.style;
+    if (!style) {
+        return;
+    }
+    
+    MGLShapeSource *source = [[MGLShapeSource alloc] initWithIdentifier:[NSUUID UUID].UUIDString URL:url options:nil];
+    [self.mapView.style addSource:source];
+    
+    NSString *pointIdentifier = [NSString stringWithFormat:@"%@ marker", source.identifier];
+    MGLSymbolStyleLayer *pointLayer = [[MGLSymbolStyleLayer alloc] initWithIdentifier:pointIdentifier source:source];
+    pointLayer.iconImageName =
+        [NSExpression expressionWithFormat:@"mgl_join({%K, '-', CAST(TERNARY(%K = 'small', 11, 15), 'NSString')})",
```

Comment:
This assumes Maki icons at the same sizes as the ones in Mapbox-designed styles such as Streets and Light.